File: shared/frontend-client/src/types/order.ts
export type OrderSource = "kiosk_classic" | "kiosk_guided" | "kiosk_premium" | "ai_agent" | "mcp";

export type OrderStatus = "draft" | "pending_payment" | "paid" | "completed" | "cancelled" | "failed";

export interface SelectedOptionInput {
  group_id: string;
  choice_ids: string[];
}

export interface OrderCreateItem {
  menu_item_id: string;
  quantity: number;
  selected_options: SelectedOptionInput[];
}

export interface OrderCreateRequest {
  source: OrderSource;
  items: OrderCreateItem[];
  simulate_failure?: boolean;
}

export interface OrderSelectedChoiceRead {
  id: string;
  name: string;
  price_delta: number;
}

export interface OrderSelectedOptionRead {
  group_id: string;
  group_name: string;
  choices: OrderSelectedChoiceRead[];
}

export interface OrderItemRead {
  id: number;
  menu_item_id: string;
  name_snapshot: string;
  unit_price: number;
  quantity: number;
  selected_options: OrderSelectedOptionRead[];
  line_total: number;
}

export interface OrderRead {
  id: number;
  user_id: string;
  source: OrderSource;
  status: OrderStatus;
  subtotal_amount: number;
  discount_amount: number;
  total_amount: number;
  created_at: string;
  updated_at: string;
  items: OrderItemRead[];
}

File: backend/app/app/schemas/order.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from app.models.enums import OrderSource, OrderStatus


class SelectedOptionInput(BaseModel):
    group_id: str
    choice_ids: list[str]


class OrderCreateItem(BaseModel):
    menu_item_id: str
    quantity: int = Field(gt=0)
    selected_options: list[SelectedOptionInput] = Field(default_factory=list)


class OrderCreateRequest(BaseModel):
    source: OrderSource
    items: list[OrderCreateItem] = Field(min_length=1)


class OrderItemRead(BaseModel):
    id: int
    menu_item_id: str
    name_snapshot: str
    unit_price: int
    quantity: int
    selected_options: list[dict[str, Any]]
    line_total: int


class OrderRead(BaseModel):
    id: int
    user_id: str
    source: OrderSource
    status: OrderStatus
    subtotal_amount: int
    discount_amount: int
    total_amount: int
    created_at: datetime
    updated_at: datetime
    items: list[OrderItemRead] = Field(default_factory=list)
